src/domain/experiment.py

```diff
@@ -207,29 +226,121 @@ class OtherLanguagesRandomCutoffSeedDictionaryAblationExperiment(RandomCutoffSee
         'source_language': ['en'],
         'target_language': ['et', 'fa', 'lv', 'vi']
     }
+
+    def __init__(self, base_config):
+        super().__init__(base_config)
+
+    def get_sbatch_args(self, run_params):
+        return {'time': '0-1:00'}
+
+
+class OtherLanguagesRandomSeedDictionaryAblationExperiment(OriginalExperiment):
+    EXPERIMENT_NAME = 'other_languages_random_seed_dictionary_ablation'
+    LANGUAGE_PARAMS = {'source_language': ['en'], 'target_language': ['et', 'fa', 'lv', 'vi']}
+    CHANGING_PARAMS = {'seed_dictionary_method': ['random_raw']}
+
+    def __init__(self, base_config):
+        super().__init__(base_config)
+
+    def get_sbatch_args(self, run_params):
+        return {'time': '0-1:00'}
+
+
+class OtherLanguagesRandomCutoffSeedDictionaryAblationExperiment(OriginalExperiment):
+    EXPERIMENT_NAME = 'other_languages_random_cutoff_seed_dictionary_ablation'
+    LANGUAGE_PARAMS = {'source_language': ['en'], 'target_language': ['et', 'fa', 'lv', 'vi']}
+    CHANGING_PARAMS = {'seed_dictionary_method': ['random_cutoff']}
+
+    def __init__(self, base_config):
+        super().__init__(base_config)
+
+    def get_sbatch_args(self, run_params):
+        return {'time': '0-1:00'}
+
+
+class GridSearchExperiment(OriginalExperiment):
+    def __init__(self, base_config):
+        super().__init__(base_config)
+
+    def __get_run_params(self, run):
+        if len(self.CHANGING_PARAMS) == 1:
+            return run.data.params[list(self.CHANGING_PARAMS)[0]]
+        else:
+            return tuple(run.data.params[param_name] for param_name in self.CHANGING_PARAMS)
+
+    def aggregate_runs(self):
+        runs = self.get_runs()
+        accuracies = defaultdict(lambda: defaultdict(list))
+        times = defaultdict(lambda: defaultdict(list))
+        num_iters = defaultdict(lambda: defaultdict(list))
+        avg_iter_times = defaultdict(lambda: defaultdict(list))
+        for run in runs:
+            if self._is_a_valid_run(run):
+                minutes = max(((run.info.end_time - run.info.start_time) // 60 // 60) % 60, 1)
```

Comment:
Why not use datetime object ?

Reply:
I only added the max with 1 to ensure no zero division error

Reply:
Yeah, but this logic of changing into minutes is already managed by the class datetime.